cpp/dolfinx/la/SparsityPattern.cpp

```diff
@@ -358,12 +381,16 @@ void SparsityPattern::assemble()
   _diagonal = std::make_shared<graph::AdjacencyList<std::int32_t>>(
       std::move(adj_data), std::move(adj_offsets));
 
-  // Column map increased due to received rows from other processes (see above)
+  // Column map increased due to received rows from other processes (see
+  // above)
   LOG(INFO) << "Column ghost size increased from "
             << _index_maps[1]->ghosts().size() << " to " << ghosts1.size()
             << "\n";
 
-  _index_maps[1] = std::make_shared<common::IndexMap>(
+  // FIXME: Do this in a more efficient way -
+  // dolfinx::MPI::compute_graph_edges is expensive. Should already have
+  // the required information on neighbour ranks
+  _column_map = std::make_shared<common::IndexMap>(
```

Comment:
Need to re-use already know ownership data. `dolfinx::MPI::compute_graph_edges` is potentially expenses and should only be used if there is not other way to compute ranks in the (forward) neighbourhood.

Reply:
This is a separate issue? Could we get this PR merged and then @IgorBaratta and I will submit another PR addressing this.